packages/web-components/src/components/masthead/masthead-global-bar.ts

```diff
@@ -20,17 +24,80 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  * @element dds-masthead-global-bar
  */
 @customElement(`${ddsPrefix}-masthead-global-bar`)
-class DDSMastheadGlobalBar extends FocusMixin(LitElement) {
+class DDSMastheadGlobalBar extends FocusMixin(HostListenerMixin(LitElement)) {
+  /**
+   * Media query listener handler.
+   */
+  private _mqHandler;
+
+  /**
+   * The name of the custom event fired after the seach is toggled.
+   */
+  static get eventToggle() {
+    return `${ddsPrefix}-masthead-search-toggled`;
+  }
+
   /**
    * The shadow slot this action bar should be in.
    */
   @property({ reflect: true })
   slot = 'profile';
 
+  /**
+   * Search bar opened flag.
+   */
+  @internalProperty()
+  private _openedSearch = false;
+
+  /**
+   * Medium breakpoint flag.
+   */
+  @internalProperty()
+  private _responsiveMd = false;
+
+  /**
+   * Handles toggle event from the search component.
+   *
+   * @param event The event.
+   */
+  @HostListener('parentRoot:eventToggle')
+  // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
+  private _handleSearchToggle = (event: Event) => {
+    this._openedSearch = (event as CustomEvent).detail.active;
+  };
+
+  /**
+   * Handles media query event.
+   *
+   * @param event The event.
+   */
+  private _handleMediaQuery(event: Event) {
+    this._responsiveMd = (event as MediaQueryListEvent).matches;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    const _mqMedium = this.ownerDocument?.defaultView?.matchMedia(`(max-width: ${breakpoints.md.width})`);
+    this._responsiveMd = _mqMedium ? _mqMedium.matches : false;
+
+    this._mqHandler = on(_mqMedium, 'change', this._handleMediaQuery.bind(this));
+  }
+
+  disconnectedCallback() {
+    if (this._mqHandler) {
+      this._mqHandler.release();
```

Comment:
```suggestion
      this._mqHandler = this._mqHandler.release();
```

This codebase nulls out handle properties when they are no longer in use.